### PR TITLE
fix: compiling with invalid file causes panic

### LINKED ISSUE #1532: Compiling with invalid file causes compiler panic
### Community Note

> Please vote by adding a 👍 reaction to the issue to help us prioritize.
> If you are interested to work on this issue, please leave a comment.

### I tried this:

`wing compile -t sim non_existent_file.w`

### I expected this:

A regular compiler diagnostic / error message

### Instead, this happened:

Compiler panicked:

```
wing compile -t sim blah.w
thread '<unnamed>' panicked at 'Source path must be a file', libs/wingc/src/lib.rs:222:5
stack backtrace:
wasm://wasm/00874ab6:1


RuntimeError: unreachable
    at std::panicking::rust_panic_with_hook::hc5c73bd02fe928d4 (wasm://wasm/00874ab6:wasm-function[185]:0xadf1d)
    at core::panicking::panic_fmt::h1024bc10a158814e (wasm://wasm/00874ab6:wasm-function[1148]:0x14f629)
    at wingc_compile (wasm://wasm/00874ab6:wasm-function[155]:0xa440d)
    at wingc_compile.command_export (wasm://wasm/00874ab6:wasm-function[1280]:0x1522c4)
    at Object.invoke (/Users/chrisr/.nvm/versions/node/v18.12.1/lib/node_modules/winglang/dist/wingc.js:74:37)
    at Command.compile (/Users/chrisr/.nvm/versions/node/v18.12.1/lib/node_modules/winglang/dist/commands/compile.js:84:40)

Node.js v18.12.1
```

### Component:

Compiler

### Wing Version:

0.5.36

### Node.js Version:

18.12.1

### Platform(s):

MacOS

### Anything else?

We should be reporting a diagnostic instead of doing an assert in the compiler

### PR DESCRIPTION
**Pull request by Chriscbr**
Fixes #1532

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

File: libs/wingc/src/lib.rs [hunk 1]
```diff
@@ -219,11 +219,32 @@ fn add_builtin(name: &str, typ: Type, scope: &mut Scope, types: &mut Types) {
...
+		}]);
+	}
+
+	if source_path.extension().unwrap() != "w" {
+		return Err(vec![Diagnostic {
+			message: format!("Source file must have .w extension: {}", source_path.display()),
+			span: None,
+			level: DiagnosticLevel::Error,
+		}]);
+	}
```
Comment (MarkMcCulloh) [inline libs/wingc/src/lib.rs:247]: What's the importance of this requirement? Just preventing accidental compilation of random stuff?

**Review by MarkMcCulloh — APPROVED** [review]

File: libs/wingc/src/lib.rs (hunk 1 above)
Comment (Chriscbr) [inline libs/wingc/src/lib.rs:247]:
```suggestion
```

File: libs/wingc/src/lib.rs [hunk 2]
```diff
@@ -219,11 +219,24 @@ fn add_builtin(name: &str, typ: Type, scope: &mut Scope, types: &mut Types) {
 }
 
 pub fn compile(source_path: &Path, out_dir: Option<&Path>) -> Result<CompilerOutput, Diagnostics> {
-	assert!(source_path.is_file(), "Source path must be a file");
-	assert!(
-		source_path.extension().unwrap() == "w",
-		"Source file must have .w extension",
-	);
+	if !source_path.exists() {
```
Comment (staycoolcall911) [inline libs/wingc/src/lib.rs:222]: Great! Can we add a test for that?

Comment (Chriscbr) [reply libs/wingc/src/lib.rs:222]: https://github.com/winglang/wing/pull/1552

**Pull request merged**